lib/Cake/Test/Case/Controller/Component/CookieComponentTest.php

```diff
@@ -790,9 +752,99 @@ public function testDeleteChildrenNotExist() {
 	}
 
 /**
+ * Test that deleting a child entry of the cookie array works
+ *
+ * @return void
+ */
+	public function testDeleteChildEntry() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('User.name', 'user', false);
+		$this->Cookie->write('User.email', 'user@example.com', false);
```

Comment:
Thank you for the detailed explanation. I misunderstood what `Hash::flatten()` would do to the data :blush:. I'm a bit concerned that we'll be sending significantly more cookies compared to previous releases and even 2.4.2. If all those cookies are encrypted, the data that has to be passed back to the server on every request is much larger.